frontend/catalyst/from_plxpr/from_plxpr.py

```diff
@@ -240,6 +259,32 @@ def calling_convention(*args):
 }
 
 
+def apply_compiler_decompose_to_plxpr(inner_jaxpr, consts, tgatesets, ncargs):
+    """Apply the compiler-specific decomposition for a given JAXPR."""
+
+    # disable the graph decomposition optimization
+    # Why? Because for the compiler-specific decomposition we want to
+    # only decompose higher-level gates and templates that only have
+    # a single decomposition, and not do any further optimization
+    # based on the graph solution.
+    # Besides, the graph-based decomposition is not supported
+    # yet in from_plxpr for most gates and templates.
+    # TODO: Enable the graph-based decomposition
+    qml.decomposition.disable_graph()
+
+    # First perform the pre-mlir decomposition to simplify the jaxpr
+    # by decomposing high-level gates and templates
+    gate_set = COMPILER_OPERATIONS + list(set().union(*tgatesets))
```

Comment:
Shouldn't this be a set union rather than a list concatenation? Also, does it ever happen that an operator is in the target gate set but not in `COMPILER_OPERATIONS`? What do you do then?

Reply:
> Does it ever happen that an operator is in the target gate set but not in COMPILER_OPERATIONS?

yes, FTQC examples w/ custom qml.ftqc operators